WalletWasabi/WabiSabi/Backend/Rounds/Arena.cs

```diff
@@ -457,6 +457,54 @@ public async Task SignTransactionAsync(TransactionSignaturesRequest request)
 			}
 		}
 
+		public async Task<ReissueCredentialResponse> ReissuanceAsync(ReissueCredentialRequest request)
+		{
+			using (await AsyncLock.LockAsync().ConfigureAwait(false))
+			{
+				if (!Rounds.TryGetValue(request.RoundId, out var round))
+				{
+					throw new WabiSabiProtocolException(WabiSabiProtocolErrorCode.RoundNotFound, $"Round ({request.RoundId}) not found.");
+				}
+
+				if (round.Phase is not (Phase.ConnectionConfirmation or Phase.OutputRegistration))
+				{
+					throw new WabiSabiProtocolException(WabiSabiProtocolErrorCode.WrongPhase, $"Round ({round.Id}): Wrong phase ({round.Phase}).");
+				}
+
+				if (request.RealAmountCredentialRequests.Delta != 0)
+				{
+					throw new WabiSabiProtocolException(WabiSabiProtocolErrorCode.DeltaNotZero, $"Round ({round.Id}): Amount credentials delta must be zero.");
+				}
+
+				//if (request.RealVsizeCredentialRequests.Delta != 0)
+				//{
+				//	throw new WabiSabiProtocolException(WabiSabiProtocolErrorCode.DeltaNotZero, $"Round ({request.RoundId}): Vsize credential delta must be zero.");
+				//}
+
+				if (request.RealAmountCredentialRequests.Requested.Count() != ProtocolConstants.CredentialNumber)
+				{
+					throw new WabiSabiProtocolException(WabiSabiProtocolErrorCode.WrongNumberOfCreds, $"Round ({round.Id}): Incorrect requested number of amount credentials.");
+				}
+
+				if (request.RealVsizeCredentialRequests.Requested.Count() != ProtocolConstants.CredentialNumber)
+				{
+					throw new WabiSabiProtocolException(WabiSabiProtocolErrorCode.WrongNumberOfCreds, $"Round ({round.Id}): Incorrect requested number of weight credentials.");
+				}
+
+				var commitRealAmountCredentialResponse = round.AmountCredentialIssuer.PrepareResponse(request.RealAmountCredentialRequests);
+				var commitRealVsizeCredentialResponse = round.VsizeCredentialIssuer.PrepareResponse(request.RealVsizeCredentialRequests);
+				var commitZeroAmountCredentialResponse1 = round.AmountCredentialIssuer.PrepareResponse(request.ZeroAmountCredentialRequests1);
+				var commitZeroAmountCredentialResponse2 = round.AmountCredentialIssuer.PrepareResponse(request.ZeroAmountCredentialRequests2);
```

Comment:
what if we replace the number suffix by a `extra` or `additional` prefix?

Reply:
but why do we need the number suffix?

when k=2 there only need to be two zero credential requests, one for amount, one for vsize, that's all, exactly like this:

https://github.com/zkSNACKs/WalletWasabi/blob/589fa819a8dad6290135a5c7e2f8070cc0016228/WalletWasabi/WabiSabi/Client/ArenaClient.cs#L155-L164